clippy_lints/src/empty_drop.rs

```diff
@@ -0,0 +1,54 @@
+use clippy_utils::diagnostics::span_lint_and_help;
+// use clippy_utils::ty::is_type_lang_item;
+use if_chain::if_chain;
+use rustc_hir::{Body, ExprKind, Impl, ImplItemKind, Item, ItemKind, Node};
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_session::{declare_lint_pass, declare_tool_lint};
+
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for empty `Drop` implementations.
+    ///
+    /// ### Why is this bad?
+    /// Empty `Drop` implementations do nothing except hurt code readability. They also prevent the type from being destructured.
```

Comment:
This is contradicting itself. I would formulate it like this:

```suggestion
    /// Empty `Drop` implementations have no effect when dropping an instance of the type. They are most likely useless. However, an empty `Drop` implementation prevents a type from being destructured, which might be the intention behind adding the implementation as a marker. 
```

This obviously needs formatting.